src/containers/mode-tools.jsx

```diff
@@ -13,10 +13,126 @@ class ModeTools extends React.Component {
+            '_getSelectedUnpointedPoints',
+            'hasSelectedUncurvedPoints',
+            'hasSelectedUnpointedPoints',
             'handleCopyToClipboard',
-            'handlePasteFromClipboard'
+            'handleCurvePoints',
+            'handlePasteFromClipboard',
+            'handlePointPoints'
         ]);
     }
+    _getSelectedUncurvedPoints () {
+        const items = [];
+        const selectedItems = getSelectedLeafItems();
+        for (const item of selectedItems) {
+            if (!item.segments) continue;
+            for (const seg of item.segments) {
+                if (seg.selected) {
+                    const prev = seg.getPrevious();
+                    const next = seg.getNext();
+                    const isCurved =
+                        (!prev || seg.handleIn.length > 0) &&
+                        (!next || seg.handleOut.length > 0) &&
+                        (prev && next ? seg.handleOut.isColinear(seg.handleIn) : true);
+                    if (!isCurved) items.push(seg);
+                }
+            }
+        }
+        return items;
+    }
+    _getSelectedUnpointedPoints () {
+        const points = [];
+        const selectedItems = getSelectedLeafItems();
+        for (const item of selectedItems) {
+            if (!item.segments) continue;
+            for (const seg of item.segments) {
+                if (seg.selected) {
+                    if (seg.handleIn.length > 0 || seg.handleOut.length > 0) {
+                        points.push(seg);
+                    }
+                }
+            }
+        }
+        return points;
+    }
+    hasSelectedUncurvedPoints () {
+        const points = this._getSelectedUncurvedPoints();
+        return points.length > 0;
+    }
+    hasSelectedUnpointedPoints () {
+        const points = this._getSelectedUnpointedPoints();
+        return points.length > 0;
+    }
+    handleCurvePoints () {
+        let changed;
+        const points = this._getSelectedUncurvedPoints();
+        for (const point of points) {
+            const prev = point.getPrevious();
+            const next = point.getNext();
+            const noHandles = point.handleIn.length === 0 && point.handleOut.length === 0;
+            if (!prev && !next) {
+                continue;
+            } else if (prev && next && noHandles) {
+                // Handles are parallel to the line from prev to next
+                point.handleIn = prev.point.subtract(next.point)
+                    .normalize()
+                    .multiply(prev.getCurve().length / 2);
+            } else if (prev && !next && point.handleIn.length === 0) {
+                // Point is end point
+                // Direction is average of normal at the point and direction to prev point, using the
+                // normal that points out from the convex side
+                // Lenth is curve length / 2
+                const convexity = prev.getCurve().getCurvatureAt(.1) < 0 ? -1 : 1;
```

Comment:
Should this use the `getCurvatureAtTime` function here, since we are using getNormalAtTime below? Is there a difference? The documentation is kind of confusing on whether getCurvatureAt takes a physical distance or a time parameter...

Reply:
The documentation didn't mention getCurvatureAtTime, thanks! That makes more sense.